tests/test_gear_validations.py

```diff
@@ -0,0 +1,603 @@
+"""Test validation functions."""
```

Comment:
Here you are testing the behaviour of the functions, these are all integration tests, move them there and add the unit ones here